docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -55,28 +55,73 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
 ### 2. Create a controller
+The simplest way to create a controller is by using the static method `channelListController(query:)` on your `ChatClient`.
+```swift
+let controller = ChatClient.shared.channelListController(query: query)
+```
+By default, the SDK will automatically handle filtering the channels as they get created. Whenever there is a web socket event that a channel has been created, the SDK will only insert it in the channel list if it matches the query.
+
+:::note
+In cases though, where the query provided contains, extra data or custom filters, **the SDK may not be able to automatically match the filter query**. In this case, you will need to provide a filtering closure.
+:::
+
+#### Filtering with extra data
+First we need to disable the Channel auto-filtering. We can do that by turning the `isChannelAutomaticFilteringEnabled` in your `ChatClient` configuration, to `false`.
+```swift
+extension ChatClient {
```

Comment:
The idea in this section is to not mention yet the `isChannelAutomaticFilteringEnabled`, only in the end, in a `#### Manual Filtering` section. The idea of this section is to show an example where automatic filtering is enabled, and only the extra data property is used in the closure. Example:

Here is an example to filter
```swift
let query = ChannelListQuery(filter: .and([
    .containMembers(userIds: [userCredentials.id])),
    .equal(.type, to: .messaging),
    .equal("myCustomBooleanKey", to: true),
])

let controller = ChatClient.shared.channelListController(query: query, filter: { channel in
      channel.extraData["myCustomBooleanKey"]?.boolValue == true
}
```

👆 Here, we should explain that only the extra data needs to be in the closure